File: AI_Agents/frontend/api/test_connection.js
export const config = { runtime: 'edge' };

export default async function handler(req) {
  if (req.method !== 'POST') return new Response('Method Not Allowed', { status: 405 });
  
  try {
    const payload = await req.json();
    const conn_type = payload.type;
    const config = payload.config || {};
    
    if (conn_type === 'llm') {
      const provider = (config.provider || '').toLowerCase();
      const apiKey = config.api_key || '';
      const apiUrl = config.api_url || '';
      
      if (provider === 'openai' || provider === 'groq') {
        const url = apiUrl || (provider === 'groq' ? 'https://api.groq.com/openai/v1/models' : 'https://api.openai.com/v1/models');
        const resp = await fetch(url, { headers: { 'Authorization': `Bearer ${apiKey}` } });
        if (resp.ok) return Response.json({ status: 'success', message: `${provider.toUpperCase()} API connected!` });
        else {
           let err = await resp.text();
           return Response.json({ status: 'error', message: `HTTP ${resp.status}: ${err}` });
        }
      }
      return Response.json({ status: 'error', message: 'Unsupported provider for Edge' });
    }
    else if (conn_type === 'source') {
      const provider = (config.provider || '').toLowerCase();
      if (provider === 'jira') {
        let domain = (config.domain_url || '').replace(/\/+$/, "");
        if (!domain.startsWith('http')) domain = `https://${domain}`;
        const email = config.email || '';
        const token = config.api_token || '';
        
        const url = `${domain}/rest/api/3/myself`;
        const auth = btoa(`${email}:${token}`);
        const resp = await fetch(url, { headers: { 'Authorization': `Basic ${auth}`, 'Accept': 'application/json' }});
        if (resp.ok) return Response.json({ status: 'success', message: 'Jira Cloud connected successfully' });
        else {
           let errText = await resp.text();
           try { const j = JSON.parse(errText); if (j.errorMessages) errText = j.errorMessages[0]; } catch(e){}
           if (resp.status === 401) errText = "Invalid API Token or Email. Ensure you use an Atlassian API Token generated from id.atlassian.com, NOT your login password.";
           return Response.json({ status: 'error', message: `HTTP ${resp.status}: ${errText}` });
        }
      }
    }
    return Response.json({ status: 'error', message: 'Unknown connection request' });
  } catch (e) {
    return Response.json({ status: 'error', message: e.message });
  }
}

File: AI_Agents/tools/test_connection.py
import sys
import json
import requests
import traceback

def test_llm(config):
    provider = config.get("provider", "").lower()
    api_key = config.get("api_key", "")
    api_url = config.get("api_url", "")
    
    if not provider:
        return {"status": "error", "message": "Provider is required"}
    
    try:
        # Minimal ping logic
        if provider in ["openai", "groq"]:
            url = api_url or "https://api.openai.com/v1/models"
            if provider == "groq" and not api_url:
                url = "https://api.groq.com/openai/v1/models"
                
            headers = {"Authorization": f"Bearer {api_key}"}
            resp = requests.get(url, headers=headers, timeout=10)
            if resp.status_code == 200:
                return {"status": "success", "message": f"{provider.capitalize()} API connected successfully"}
            else:
                return {"status": "error", "message": f"HTTP {resp.status_code}: {resp.text}"}
                
        elif provider == "ollama":
            url = api_url or "http://localhost:11434/api/tags"
            resp = requests.get(url, timeout=5)
            if resp.status_code == 200:
                return {"status": "success", "message": "Ollama engine is available"}
            else:
                return {"status": "error", "message": f"HTTP {resp.status_code}: {resp.text}"}

        elif provider == "grok":
            return {"status": "success", "message": "Grok placeholder validation (Success)"}
            
        else:
            return {"status": "error", "message": f"Unsupported LLM provider: {provider}"}
            
    except requests.exceptions.RequestException as e:
        return {"status": "error", "message": f"Connection failed: {str(e)}"}
        
def test_source(config):
    provider = config.get("provider", "").lower()
    if provider == "jira":
        domain = config.get("domain_url", "").strip("/")
        if not domain.startswith("http"):
            domain = f"https://{domain}"
            
        email = config.get("email", "")
        token = config.get("api_token", "")
        
        if not domain or not email or not token:
            return {"status": "error", "message": "Jira requires domain_url, email, and api_token"}
            
        url = f"{domain}/rest/api/3/myself"
        headers = {
            "Accept": "application/json"
        }
        try:
            resp = requests.get(url, headers=headers, auth=(email, token), timeout=10)
            if resp.status_code == 200:
                return {"status": "success", "message": "Jira Cloud connected successfully"}
            else:
                try:
                    err_text = resp.json().get("errorMessages", [resp.text])[0]
                except:
                    err_text = resp.text
                if resp.status_code == 401:
                    err_text = "Invalid API Token or Email. Ensure you are using an Atlassian API Token generated from id.atlassian.com, NOT your login password."
                return {"status": "error", "message": f"HTTP {resp.status_code}: {err_text}"}
        except requests.exceptions.RequestException as e:
             return {"status": "error", "message": f"Jira Connection failed: {str(e)}"}
             
    elif provider == "ado":
        return {"status": "error", "message": "ADO validation not strictly implemented yet"}
    else:
        return {"status": "error", "message": f"Unsupported source provider: {provider}"}

def main():
    if len(sys.argv) < 2:
        print(json.dumps({"status": "error", "message": "Missing JSON payload"}))
        sys.exit(1)
        
    try:
        payload = json.loads(sys.argv[1])
        conn_type = payload.get("type")
        config = payload.get("config", {})
        
        if conn_type == "llm":
            result = test_llm(config)
        elif conn_type == "source":
            result = test_source(config)
        else:
            result = {"status": "error", "message": f"Unknown connection type: {conn_type}"}
            
        print(json.dumps(result))
    except Exception as e:
        print(json.dumps({"status": "error", "message": f"Internal Error: {str(e)}", "trace": traceback.format_exc()}))

if __name__ == "__main__":
    main()
